pgmpy/readwrite/ProbModelXML.py

```diff
@@ -483,32 +487,32 @@ def add_node(self, variable):
         variable_name = variable.attrib['name']
         self.probnet['Variables'][variable_name] = {}
         self.probnet['Variables'][variable_name]['type'] = variable.attrib['type']
-        self.probnet['Variables'][variable_name]['roles'] = variable.attrib['roles']
+        self.probnet['Variables'][variable_name]['role'] = variable.attrib['role']
         if variable.find('Comment'):
             self.probnet['Variables'][variable_name]['Comment'] = variable.find('Comment').text
         if variable.find('Coordinates'):
             self.probnet['Variables'][variable_name]['Coordinates'] = variable.find('Coordinates').attrib
-        if variable.xpath('AdditionalProperties'):
-            for prop in variable.find('AdditionalProperties').iterchildren():
+        if variable.find('AdditionalProperties/Property'):
+            for prop in variable.findall('AdditionalProperties/Property'):
                 self.probnet['Variables'][variable_name]['AdditionalProperties'][prop.attrib['name']] = \
                     prop.attrib['value']
-        if not variable.xpath('States'):
+        if not variable.find('States/State'):
             warnings.warn("States not available for node: " + variable_name)
         else:
-            self.probnet['Variables'][variable_name]['States'] = {state.attrib['name']: {prop.attrib['name']: prop.attrib['value'] for prop in state.find('AdditionalProperties')} for state in variable.find('States').iterchildren()}
+            self.probnet['Variables'][variable_name]['States'] = {state.attrib['name']: {prop.attrib['name']: prop.attrib['value'] for prop in state.findall('AdditionalProperties/Property')} for state in variable.findall('States/State')}
 
     def add_edge(self, edge):
         var1 = edge.attrib['var1']
         var2 = edge.attrib['var2']
         self.probnet['edges'][(var1, var2)] = {}
         self.probnet['edges'][(var1, var2)]['directed'] = edge.attrib['directed']
-        if edge.xpath('Comment'):
+        if edge.find('Comment'):
         #TODO: check for the case of undirected graphs if we need to add to both elements of the dic for a single edge.
-            self.probnet['edges'][(var1, var2)]['Comment'] = edge.xpath('Comment').text
-        if edge.xpath('Label'):
-            self.probnet['edges'][(var1, var2)]['Label'] = edge.xpath('Label').text
-        if edge.xpath('AdditionalProperties'):
-            for prop in edge.xpath('AdditioanlProperties')[0].iterchildren():
+            self.probnet['edges'][(var1, var2)]['Comment'] = edge.find('Comment').text
+        if edge.find('Label'):
```

Comment:
If  i use  
           if edge.find('Label'): 
then it is not getting inside if statement
but if i use 
                 if edge.find('Label') is not None: 
then it is getting inside if statement. 
What is the problem ?
http://stackoverflow.com/questions/3901144/difference-between-if-x-and-if-x-is-not-none
I think it is better to use is not None also.